src/main/java/me/ryanhamshire/GriefPrevention/EntityEventHandler.java

```diff
@@ -335,11 +335,11 @@ public void onFallingBlockEnterPortal(EntityPortalEnterEvent event)
     //Don't let people drop in TNT through end portals
     //Necessarily this shouldn't be an issue anyways since the platform is obsidian...
     @EventHandler(ignoreCancelled = true)
-    void onTNTExitPortal(EntityPortalExitEvent event)
+    void onTNTExitPortal(EntityPortalEnterEvent event)
     {
         if (event.getEntityType() != EntityType.PRIMED_TNT)
             return;
-        if (event.getTo().getWorld().getEnvironment() != Environment.THE_END)
+        if (event.getLocation().getBlock().getType() == Material.END_PORTAL)
```

Comment:
Would this handle cases of end portal blocks that don't actually end up teleporting an entity/are plugin-handled/end portals in non-default worlds?

Reply:
Yes, will deal with all of the above. 

Using the EntityEnterPortalEvent, you can only determine the location of entry and what block it is. 

It shouldn't interfere with any portal plugins, however would probably be best to be configurable for future references if the server admins would like it to be close to the vanilla state.

Reply:
I think Robo's more asking "will this result in TNT that is not teleporting being removed" and I believe the answer is yes, though I'm not sure that it matters much. We could swap to the EntityTeleportEvent instead and see if it's from an end world to a non-end world with the correct cause.

Reply:
Yes, it would although as you said I don't believe there are many problems that could arise as a result of this change. In most cases, if the End hasn't been generated, the TNT will just drop into the lava at the end portal structure.

In terms of clashes with other possible plugins which use portals, they would probably there own instances to remove other entities from teleporting.

I've attempted the EntityTeleportEvent before this and it would still crash the server.

Reply:
It crashes even when cancelled? That seems shocking, honestly. Either way, yeah, I don't see this implementation as being problematic.

Reply:
Yeah, I tried that method first before attempting this one. Obviously the destination would be a lot better to use but it would also crash the server.